stl/inc/expected

```diff
@@ -722,6 +1023,24 @@ public:
     }
 
 private:
+    // These overloads force copy elision from the invoke call into _Value
+    template <class _Fn, class _Ux>
+    constexpr expected(_Construct_expected_from_invoke_result_tag, _Fn&& _Func, _Ux&& _Arg) noexcept(
+        is_nothrow_constructible_v<_Ty, invoke_result_t<_Fn, _Ux>>)

```

Comment:
Do we test internal parts of the library which are not observable to client code? I can see how this might affect codegen in some cases, but since these constructors are only ever called from `transform` and `transform_error`, which are not marked as `noexcept`, I don't think it's observable to client code.

Reply:
> Do we test internal parts of the library which are not observable to client code? I can see how this might affect codegen in some cases, but since these constructors are only ever called from `transform` and `transform_error`, which are not marked as `noexcept`, I don't think it's observable to client code.

No we don't. I was thinking code elsewhere might depend on this conditional noexcept; since that's not the case I withdraw my suggestion for test coverage.

> Also, making this change seems to cause issues with `/permissive` mode. What's the best resolution for that?

It's a failure to properly defer materialization of the prvalue `Immovable` returned by `immov`. I _think_ this is a known MSVC bug. I suggest copying the definition of `is_permissive` from `<range_algorithm_support.hpp>`:

https://github.com/microsoft/STL/blob/f8b8ad6dea901d360d8165cc6719de47ca9d9ca2/tests/std/include/range_algorithm_support.hpp#L26-L45

into your test, and wrapping the four test cases that use `immov` in `if constexpr (!is_permissive) // TRANSITION, FIXME`. I'll dig up a bug number to replace the `FIXME`s so we can merge this.

Reply:
I've internally filed the deferred materialization bug as VSO-1734935, so you can guard the "workaround"s with `// TRANSITION, VSO-1734935`.